src/services/smart-action-hook.js

```diff
@@ -36,7 +36,8 @@ class SmartActionHook {
     if (typeof hook !== 'function') throw new Error('hook must be a function');
 
     // Call the user-defined load hook.
-    const result = await hook({ record: { id: recordId, ...record }, fields: fieldsForUser });
+    if (!record) record = { id: recordId };
```

Comment:
Great!

Could you move the comment `// Call the user-defined load hook.` back with line 40 please?
Furthermore, we could add another comment specifying `recordId` is for handling smart collection :)